hw02/PlayerController.cs

```diff
@@ -18,34 +22,65 @@ 4. Jump Speed as a float
-    C# document (PlayerController.cs)?
+    C# document (PlayerController.cs)? 
     */
-    
+
+    public string characterName;
+    public float speed;
+    public float gravityValue;
+    public float jumpSpeed;
+    public PlayerController myPlayerController;
+
     // Start is called before the first frame update
-    void Start()
+    private void Start()
     {
+        // Answer: myPlayerController = GameObject.Find("Player").GetComponent<PlayerController>();
         // TODO: Problem 3: Describe when you would put code in the Start() function instead of the Update() function. 
+        // Answer: The part of the game when you want the code to run when the game start but not want they update each frame.
         // What is the difference between Start() and Update()?
+        // Answer: The Start() will only run when the game start, the Update() will run each frame during the game running to make sure everything update ontime.
     }
 
     // Update is called once per frame
-    void Update()
+    private void Update()
     {
         // TODO: Problem 4: What is an if statement? Write a couple of if statements to move the player with the WASD keys. 
+        // Answer: A if statement is you run the script only when the if condition satisfied.
         // We want to transform the player's position along the vertical and horizontal axis then multiply it by 
         // the corresponding variables we defined earlier. Lastly, we will need to multiply it all by Time.DeltaTime
+
+        if (Input.GetKey(KeyCode.W))
+        {
+            transform.Translate(new Vector3(0, 0, 5) * Time.deltaTime);
+        }
+        else if (Input.GetKey(KeyCode.S))
+        {
+            transform.Translate(new Vector3(0, 0, -5) * Time.deltaTime);
+        }
+        //Debug.Log(Time.deltaTime);
+
+        if (Input.GetKey(KeyCode.A))
+        {
+            transform.Translate(new Vector3(-5, 0, 0) * Time.deltaTime);
+        }
+        else if (Input.GetKey(KeyCode.D))
+        {
+            transform.Translate(new Vector3(5, 0, 0) * Time.deltaTime);
+        }
     }
 
     // TODO: Problem 5: What do the lines here do? Please provide inline comments and describe what each line is doing
     // What do Header, SerializeField, and Tooltip mean? 
-    [Header("General Setup Settings")]
-    [SerializeField] private InputAction movement;
-    [Tooltip("How fast player moves up and down based upon player input")][SerializeField] float controlSpeed = 30f;
-    [Tooltip("How far player moves horizontally")][SerializeField] float xRange = 10f;
-    [Tooltip("How far player moves vertically")][SerializeField] float yRange = 10f;
+    [Header("General Setup Settings")] // In the Inspector add UI with Header looking style, to help increase the readibility.
+    [SerializeField] private InputAction movement; // To make private variable to displayable and editable in Unity Inspector.
```

Comment:
That's correct! Header, SerializedField and tooltip are all property attributes that you can create so that others using the inspector will know what each variable means.